src/main/java/frc/robot/commands/Turret/VisionAimTurret.java

```diff
@@ -38,25 +52,47 @@ public void initialize() {
   @Override
   public void execute() {
     target = -vision.limelight.getOffsetX() + turret.getTurretAngle();
+    SmartDashboard.putNumber("oldTargetPosition", oldTargetPosition);
 
-    if (vision.limelight.hasTarget()) {
-      turret.setTurretAngle(target);
-    }
+    boolean isPressed = RobotContainer.coDriverStick.btn_A.get();
 
-    // if (shooter.isGoalHighHub()) {
-    // shooter.setGoalRPM(vision.getIdealMediumHoodRPM());
-    // } else {
-    // shooter.setGoalRPM(vision.getIdealLowerHubRPM());
-    // }
+    if (isPressed) {
```

Comment:
If you are checking to see if a button is pressed here, you are essentially emulating a while_held command. 

Think about what your goal is for having this button check. You are on the right track, just have your button check in the wrong place.